components/governance/org.wso2.carbon.governance.taxonomy/src/main/java/org/wso2/carbon/governance/taxonomy/services/TaxonomyServicesImpl.java

```diff
@@ -25,92 +27,125 @@
 import org.wso2.carbon.governance.taxonomy.util.CommonUtils;
 import org.wso2.carbon.registry.common.services.RegistryAbstractAdmin;
 import org.wso2.carbon.registry.core.exceptions.RegistryException;
+import org.wso2.carbon.user.api.UserStoreException;
 import org.xml.sax.SAXException;
 import java.io.IOException;
 import javax.xml.parsers.ParserConfigurationException;
 import javax.xml.xpath.XPathExpressionException;
 
 public class TaxonomyServicesImpl extends RegistryAbstractAdmin implements ITaxonomyServices {
-
+    private static final Log log = LogFactory.getLog(TaxonomyServicesImpl.class);
     /**
      * This method will add user defined taxonomy into taxonomy.xml
      *
      * @param payload String
      * @return boolean
-     * @throws RegistryException
+     * @throws Exception
      */
     @Override public boolean
-    addTaxonomy(String payload) throws RegistryException {
+    addTaxonomy(String payload) throws Exception {
```

Comment:
Thank you Niranjan for noticing ,
We can't catch specific type exceptions inside jsp since we can't send specific type exception through stub, If we use specific exception type , there will be class not found exception for that specific exception, So i have to catch generic Exception in UI bundle. Therefore i throw the generic exception. 

Since this service consumes not only UI bundle , its better to throw registry exception here and catch generic exception in UI bundle. So this fixed with https://github.com/wso2/carbon-governance/pull/255